src/common/eth-providers/consensus-provider/consensus-provider.service.ts

```diff
@@ -340,20 +337,27 @@ export class ConsensusProviderService {
       options?.fallbackConditionCallback != undefined ? options.fallbackConditionCallback : () => true,
     ];
     const retry = retrier(this.logger, maxRetries, 100, 10000, true);
-    const res = await callback(this.rpcUrls.main)
-      .catch(rejectDelay(this.config.get('CL_BEACON_RPC_RETRY_DELAY_MS')))
-      .catch(() => retry(() => callback(this.rpcUrls.main)))
-      .catch((e: any) => {
-        if (fallbackConditionCallback(e)) {
-          if (!this.rpcUrls.backup) {
-            this.logger.warn('Backup CL RPC url not passed');
-            throw e;
+    let res;
+    for (let i = 0; i < this.rpcUrls.length; i++) {
+      if (res) break;
+      res = await callback(this.rpcUrls[i])
+        .catch(rejectDelay(this.config.get('CL_BEACON_RPC_RETRY_DELAY_MS')))
+        .catch(() => retry(() => callback(this.rpcUrls[i])))
+        .catch((e: any) => {
+          if (fallbackConditionCallback(e)) {
+            if (!this.rpcUrls.length) {
```

Comment:
`this.rpcUrls` will contain at least the first one endpoint, so the condition looks wrong.

Reply:
fixed